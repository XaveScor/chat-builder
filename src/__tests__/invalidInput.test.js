import {createPage, questionPhrase, input, Stop, ConformsForm, ParseError} from '..'
import * as React from 'react';
import * as renderer from 'react-test-renderer';

it('hide question on invalid answer', done => {
    const page = createPage()

    page.use((_, props) => {
        return {
            steps: [
                {
                    ...questionPhrase,
                    input,
                    question: 'Question',
                    error: 'Error',
                    parseAnswer: answer => new ParseError('Error'),
                }
            ],
            nextPage: Stop,
        }
    })

    const tree = renderer
        .create(<ConformsForm page={page} />)

    setTimeout(() => {
        const inputInstance = tree.root.findByType(input)
        inputInstance.props.onChange('asd')
        inputInstance.props.onSubmit()
        
        setTimeout(() => {
            const res = tree.toJSON()
            expect(res).toMatchSnapshot()
            done()
        }, 1000)
    }, 1000)
})