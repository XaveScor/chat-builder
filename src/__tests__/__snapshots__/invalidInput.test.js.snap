// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hide question on invalid answer 1`] = `
Array [
  <p
    className="robot"
  >
    Question
  </p>,
  <p
    className="answer"
  >
    asd
  </p>,
  <p
    className="error"
  >
    Error
  </p>,
  <input
    onChange={[Function]}
    value="asd"
  />,
  <input
    onClick={[Function]}
    type="submit"
  />,
]
`;
